src/pages/ImportAnalysis.js

```diff
@@ -0,0 +1,419 @@
+                iconName: 'copySolid',
+                title: 'Start with a template',
+                detail: 'Clone from one of our template workspaces that has analyses ready for use',
+              }),
+            h(ChoiceButton, {
+              onClick: () => setMode('existing'),
+              iconName: 'fileSearchSolid',
+              title: 'Start with an existing workspace',
+              detail: 'Select one of your workspaces',
+              disabled: !userHasBillingProjects,
+            }),
+            h(ChoiceButton, {
+              onClick: () => setIsCreateOpen(true),
+              iconName: 'plus-circle',
+              title: 'Start with a new workspace',
+              detail: 'Set up an empty workspace that you will configure for analysis',
+              'aria-haspopup': 'dialog',
+              disabled: !userHasBillingProjects,
+            }),
+            isCreateOpen &&
+              h(NewWorkspaceModal, {
+                requiredAuthDomain: authorizationDomain,
+                customMessage: importMayTakeTime && importMayTakeTimeMessage,
+                requireEnhancedBucketLogging: requireProtectedWorkspace,
+                onDismiss: () => setIsCreateOpen(false),
+                onSuccess: (w) => {
+                  setMode('existing');
+                  setIsCreateOpen(false);
+                  setSelectedWorkspaceId(w.workspaceId);
+                  refreshWorkspaces();
+                  onImport(w);
+                },
+              }),
+          ]);
+        },
+      ]
+    ),
+    isCloneOpen &&
+      h(NewWorkspaceModal, {
+        cloneWorkspace: _.find({ workspace: selectedTemplateWorkspaceKey }, workspaces),
+        title: `Clone ${selectedTemplateWorkspaceKey.name} and Import Data`,
+        buttonText: 'Clone and Import',
+        customMessage: importMayTakeTime && importMayTakeTimeMessage,
+        onDismiss: () => setIsCloneOpen(false),
+        onSuccess: (w) => {
+          setMode('existing');
+          setIsCloneOpen(false);
+          setSelectedWorkspaceId(w.workspaceId);
+          refreshWorkspaces();
+          onImport(w);
+        },
+      }),
+    (isImporting || loadingWorkspaces) && spinnerOverlay,
+  ]);
+};
+
+// This method identifies whether a workspace qualifies as protected.
+// 'Protected' here means that it has enhanced logging - either on its own or because it has an auth domain.
+// For now this also means only GCP workspaces are included.
+export const isProtectedWorkspace = (workspace) => {
+  if (!isGoogleWorkspace(workspace)) {
+    return false;
+  }
+  return !!workspace.workspace.bucketName && workspace.workspace.bucketName.startsWith('fc-secure');
+};
+
+// This method fetches the raw file from the URL, then writes it to the
+// workspace bucket, notebooks folder.
+export const importAnalysisFile = (namespace, name, url, filename, runtime) => {
+  return async () => {
+    if (!isRuntimeToolLabel(runtime)) {
```

Comment:
this seems pretty confusing. This method in particular would benefit from types. I'm guessing this `runtime` actually a `ToolLabel` and not a `Runtime`